desc/compute/_profiles.py

```diff
@@ -144,10 +312,15 @@ def _gradpsi_mag(params, transforms, profiles, data, **kwargs):
     transforms={},
     profiles=["pressure"],
     coordinates="r",
-    data=[],
+    data=["Te", "ne", "Ti", "Zeff"],
 )
 def _p(params, transforms, profiles, data, **kwargs):
-    data["p"] = profiles["pressure"].compute(params["p_l"], dr=0)
+    if profiles["pressure"] is not None:
+        data["p"] = profiles["pressure"].compute(params["p_l"], dr=0)
+    else:
+        data["p"] = elementary_charge * (
+            data["ne"] * data["Te"] + data["Ti"] * data["ne"] / data["Zeff"]
```

Comment:
This is correct. The number of electrons should be equal to the number of ions * effective charge of each ion, so `ne = sum(ni_k) * Zeff` where `Zeff = sum(Zi_k*ni_k)/sum(ni_k)` for `k` different ion species.
The decision to use Zeff as opposed to ion density is based on the following:
- This seems to be a more common convention
- ion density is difficult to measure, so in practice it is usually estimated based on electron density and a rough estimate of Zeff
- Using ion density by itself wouldn't allow us to account for impurities which can be important for neoclassical and turbulent transport.
- Similarly, if we wanted to properly handle impurities we'd need profiles for multiple ions as well as their charge state which gets way more complicated
- Using Zeff like this guarantees global quasi-neutrality. Allowing the user to specify electron and ion density independently could result in them giving a non-neutral plasma in which case all of our equations are super wrong.

Reply:
Thanks for the explanation. As one counter-point from my experience, SFINCS uses the density profile for each species instead of Zeff. But it could be tricky for us to implement an arbitrary number of ion profiles for each species, and your point about enforcing quasi-neutrality is important. I'm content with this choice now.